docs/pages/references/directives/directives.md

```diff
@@ -1,19 +1,157 @@
 
+Both types share some some common features:
+
+*   Application images are declared with `dimg` directive, and artifact images with `artifact` directive.
+*   Base image for an application or artifact image is declared with `from` directive.
+
 Основное различие при использовании разного синтаксиса dappfile в части описания образов - в YAML синтаксисе используется линейное описание образов и отсутствует вложенность, в Ruby синтаксисе же, можно использовать директиву `dimg_group` и описывать вложенные контексты (см. ниже.).
 
+A major difference is that YAML syntax is linear and has no nesting, while Ruby syntax can describe nested context with `dimg_group` directive.
 
 ## YAML синтаксис (dappfile.yml)
 
+## YAML syntax (dappfile.yml)
+
+### General features
+
+This syntax is based on YAML and has much in common with Dockerfiles and Ansible playbooks.
+
+Here is an example of a minimal `dappfile.yml`:
+
+```yaml
+dimg: example
+from: alpine
+```
+
+An image description should at least include the following:
+
+* image type and name, set with `dimg`, or `artifact` directive;
+* base image, set with `from`, `fromDimg`, or `fromDimgArtifact` directive.
+
+A dappfile can have multiple image descriptions, separated by `---`, as in YAML specification.
+For more details on this topic, see [Describing Multiple Images](#describing-multiple-images).
+
+### dimg
+
+The dimg directive starts a description for building an application image:
+
+```yaml
+dimg: [<name>[:<tag>]]
+```
+
+Name of an application image is a string, similar to the image name in Docker.
+The part after `:` sets the image tag and is usually used for versioning images.
+If absent, tag defaults to `latest`.
+
+```yaml
+dimg: frontend:latest
+
+# same 
+dimg: frontend
+```
+
+Name of an application image can be empty or equal to `~`.
+In a dappfile with multiple images only one can have an empty name.
+
+```yaml
+dimg:
+from: alpine
+---
+dimg: this_one_should_have_a_name
+from: alpine
+```
+
+An application image can have several names, set as a list in YAML syntax.
+This is equal to describing similar images with different names.
+
+```yaml
+dimg: [main-front,main-back]
+```
+
+### artifact
+
+The `artifact` directive starts a description for building an artifact image:
+
+```yaml
+artifact: <name>[:<tag>]
+from: <image>
+```
+
+Name of an artifact image is a string, similar to the image name in Docker.
+The part after `:` sets the image tag and is usually used for versioning images.
+If absent, tag defaults to `latest`.
+
+Artifact name is used in `import` directive (see [artifact](artifact.html) for details).
+Thus artifact images must have non-empty names.
+
+```
+artifact: hello_world
+from: alpine
+```
+
+### from
+
+The `from` directive sets the name of a base image to build from.
+
+```yaml
+from: <image>[:<tag>]
+```
+
+If absent, tag defaults to `latest`.
+
+### Describing Multiple Images
```

Comment:
Ths is something like https://flant.github.io/dapp/multiple_images_for_build.html 
But it's better, that original article.

maybee we will delete multiple_images_for_build.md and leave only directives.md?